src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

```diff
@@ -1186,13 +1238,66 @@ public InventoryRowModel() throws OHServiceException {
 				}
 			}
 			if (areAllMedicalsInInventory()) {
-				allRadio.setSelected(true);
 				specificRadio.setSelected(false);
 				codeTextField.setEnabled(false);
 			} else {
 				specificRadio.setSelected(true);
 				codeTextField.setEnabled(true);
-				allRadio.setSelected(false);
+			}
+		}
+
+		public InventoryRowModel(MedicalType medType) throws OHServiceException {
+			inventoryRowList.clear();
+			inventoryRowList = loadNewInventoryTableByMedicalType(medType);
+			if (!inventoryRowList.isEmpty()) {
+				for (MedicalInventoryRow invRow : inventoryRowList) {
+					addMedInRowInInventorySearchList(invRow);
+				}
+			}
+			if (areAllMedicalsInInventory()) {
+				specificRadio.setSelected(false);
+				codeTextField.setEnabled(false);
+			} else {
+				specificRadio.setSelected(true);
+				codeTextField.setEnabled(true);
+			}
+		}
+
+		public InventoryRowModel(boolean withNoZeroQty, MedicalType medType) throws OHServiceException {
+			inventoryRowList.clear();
+			inventoryRowList = loadNewInventoryTable(withNoZeroQty, medType);
+			if (!inventoryRowList.isEmpty()) {
+				for (MedicalInventoryRow invRow : inventoryRowList) {
+					addMedInRowInInventorySearchList(invRow);
+				}
+			} else {
+				MessageDialog.info(null, "angal.invetory.notdataforthatfilter.msg");
+			}
+			if (areAllMedicalsInInventory()) {
+				specificRadio.setSelected(false);
+				codeTextField.setEnabled(false);
+			} else {
+				specificRadio.setSelected(true);
+				codeTextField.setEnabled(true);
+			}
+		}
+
+		public InventoryRowModel(MedicalType medType, boolean withMovement) throws OHServiceException {
```

Comment:
the method parameter, `withMovement` is never used....

Reply:
It is use at line 2352